src/mod/ident.mod/ident.c

```diff
@@ -103,9 +104,14 @@ static struct dcc_table DCC_IDENTD = {
 static void ident_oidentd()
 {
   char *home = getenv("HOME");
-  char path[121], buf[(sizeof "global { reply \"\" }") + USERLEN];
-  int nbytes;
-  int fd;
+  FILE *fd;
+  long filesize;
+  char *data = NULL;
+  char path[121], line[256], buf[256], identstr[256], s[INET_ADDRSTRLEN];
```

Comment:
Please declare `s` like this:
```
#ifdef IPV6
  char s[INET6_ADDRSTRLEN];
#else
  char s[INET_ADDRSTRLEN];
#endif
```
Else `inet_ntop(..., s, ...)` in line 180 does stack overflow